File: Coffee.js
function Coffee() {
    this.empty = true;
}

Coffee.prototype.refill = function() {
    if (!this.empty) {
        throw new Error('Coffee cup is already full');
    }

    this.empty = false;
};

Coffee.prototype.drink = function() {
    if (this.empty) {
        throw new Error('Coffee cup is empty');
    }

    this.empty = true;
};

Coffee.prototype.isEmpty = function() {
    return this.empty;
};

module.exports = Coffee;

File: matheus_marabesi/src/Coffee.java

public class Coffee {

	private boolean empty = true;
	
	public boolean isEmpty() {
		return empty;
	}
	
	public boolean refill() throws Exception{
		if (!empty) {
			throw new Exception("You already have coffee to drink");
		}
		
		empty = false;
		
		return true;
	}
	
	public boolean drink() throws Exception {
		if (empty) {
			throw new Exception("There is no coffee to drink");
		}
		
		return empty = true;
	}
}
